api/envoy/service/auth/v3/attribute_context.proto

```diff
@@ -116,7 +117,13 @@ message AttributeContext {
     // The HTTP request headers. If multiple headers share the same key, they
     // must be merged according to the HTTP spec. All header keys must be
     // lower-cased, because HTTP header keys are case-insensitive.
-    map<string, string> headers = 3;
+    // Header value is encoded as string. This does not work for non-utf8 characters.
+    map<string, string> headers = 3
```

Comment:
Representing these as a map (not multimap) was a mistake (made a very very long time ago). While we're breaking compatibility, I think we should probably move to the same format as ext_proc uses, which is to use `config.core.v3.HeaderMap`, which contains a `repeated HeaderValue` for the underlying representation.

In `HeaderValue` there is already a field for `bytes raw_value`, so when the new config flag is true, always use `raw_value` instead of `value`.

Reply:
That's a great observation.
To avoid confusion, I think there should be a rejection of using the new field when the string-value headers field are used, and only allow bytes-value headers.